scripts/install

```diff
@@ -109,70 +110,103 @@ ynh_add_fpm_config
 #=================================================
 # INSTALL AND INITIALIZE COMPOSER
 #=================================================
+ynh_print_info "Installing roundcube with composer..."
 
-init_composer "$final_path"
+# Install composer.json
+cp "$final_path/composer.json-dist" "$final_path/composer.json"
+
+# Install composer
+ynh_install_composer
 
 #=================================================
 # INITIALIZE DATABASE
 #=================================================
+ynh_print_info "Initializing database..."
 
 ynh_mysql_connect_as "$db_name" "$db_pwd" "$db_name" \
-    < "${final_path}/SQL/mysql.initial.sql"
+    < "$final_path/SQL/mysql.initial.sql"
 
 #=================================================
 # CONFIGURE ROUNDCUBE
 #=================================================
+ynh_print_info "Configuring roundcube..."
+
+rc_conf="$final_path/config/config.inc.php"
 
-rc_conf="${final_path}/config/config.inc.php"
 cp ../conf/config.inc.php "$rc_conf"
 
-ynh_replace_string "#DESKEY#" "$(ynh_string_random 24)" "$rc_conf"
-ynh_replace_string "#DBUSER#" "$db_name" "$rc_conf"
-ynh_replace_string "#DBPASS#" "$db_pwd" "$rc_conf"
-ynh_replace_string "#DBNAME#" "$db_name" "$rc_conf"
+ynh_replace_string "__DESKEY__" "$(ynh_string_random 24)" "$rc_conf"
+ynh_replace_string "__DBUSER__" $db_name "$rc_conf"
+ynh_replace_string "__DBPASS__" "$db_pwd" "$rc_conf"
+ynh_replace_string "__DBNAME__" $db_name "$rc_conf"
 
 #=================================================
 # INSTALL ADDITIONAL PLUGINS
 #=================================================
+ynh_print_info "Installing additional plugins..."
 
 # Create logs and temp directories
-mkdir -p "${final_path}/logs" "${final_path}/temp"
+mkdir -p "$final_path/"{logs,temp}
 
 # Install contextmenu and automatic_addressbook plugins
 # https://plugins.roundcube.net/packages/sblaisot/automatic_addressbook
 # https://plugins.roundcube.net/packages/johndoh/contextmenu
-exec_composer "$final_path" require \
-    "johndoh/contextmenu $contextmenu_version" \
-    "sblaisot/automatic_addressbook $automatic_addressbook_version"
+ynh_composer_exec --commands="require \
+    johndoh/contextmenu $contextmenu_version \
+    sblaisot/automatic_addressbook $automatic_addressbook_version"
+
 installed_plugins+=" 'contextmenu', 'automatic_addressbook',"
 
 # Install CardDAV plugin
-if [[ $with_carddav -eq 1 ]]; then
-  install_carddav "$final_path" \
-    && installed_plugins+=" 'carddav'," \
-    || echo "Unable to install CardDAV plugin" >&2
+if [ $with_carddav -eq 1 ]
+then
+    ynh_composer_exec --commands="require roundcube/carddav $carddav_version"
+
+    carddav_tmp_config="../conf/carddav.config.inc.php"
+
+    # Look for installed and supported CardDAV servers
+    for carddav_app in "owncloud" "baikal"
```

Comment:
Maybe adding nextcloud here? Same config as owncloud.

Reply:
You're right.
The thing is I'm not fixing anything or adding any feature, just upgrading the package to last standard.

Reply:
Ok, I'm adding this on my todo list

Reply:
I was saying that for the record.
But here, it's just about adding nextcloud. So I'll do it.

Reply:
You should add nextcloud section in this file then : https://github.com/YunoHost-Apps/roundcube_ynh/blob/package_upgrade/conf/carddav.config.inc.php

Reply:
Arg... Are we sure that the same thing ?
I'm not using all that

Reply:
Me neither.
On the internet I found the same url, but I can't test it for now.

Let me some days to have time to test it on my ynh-dev.